src/example_fgen_basic/result/result_dp_manager.f90

```diff
@@ -128,8 +144,12 @@ subroutine check_index_claimed(instance_index)
             ! if it fails, the result_dp attribute will be set).
             ! So the code would be something like
             ! res = ResultNone(ResultDP(code=1, message="Index ", instance_index, " has not been claimed"))
-            print *, "Index ", instance_index, " has not been claimed"
-            error stop 1
+            ! print *, "Index ", instance_index, " has not been claimed"
+            ! error stop 1
+            ! MZ Weird thing allocatable message
```

Comment:
yep lots of mucking around. We may end up writing something like this https://gitlab.com/magicc/fgen/-/blob/switch-to-uv/packages/fgen-core/src/char_conversions.f90?ref_type=heads

or using someone else's equivalent if we can find a nice one

Reply:
ok the pros are here: https://github.com/fortran-lang/stdlib

we would either use their functions or could also just look at them and write our own versions because a) stdlib is MIT licensed so it's fine and b) we don't introduce the entire stdlib as a dependency that way